wrangler-core/src/main/java/io/cdap/wrangler/utils/ArithmeticOperations.java

```diff
@@ -388,12 +391,64 @@ public static Float min(Float f1, Float f2) {
   }
 
   /**
-   Arithmetic operation - Find the minimum of two columns.
+   * Arithmetic operation - Find the minimum of two columns.
    */
   public static BigDecimal min(BigDecimal bd1, BigDecimal bd2) {
     if (bd1 == null || bd2 == null) {
       return null;
     }
     return bd1.min(bd2);
   }
+
+  /**
+   * Arithmetic operation - calculate average of any number of columns.
+   * Returns Number - BigDecimal if any column is of type BigDecimal, double otherwise.
+   * HALF_EVEN rounding mode chosen for high precision in the long run.
+   */
+  public static Number average(Number... nums) throws DirectiveExecutionException {
+    // Note that this algorithm avoids overflow
+    Number avg = 0.0;
+    int t = 1;
+    boolean allNull = true;
+    boolean noBigDecimal = true;
+    for (Number num : nums) {
+      // Check for null input
+      if (num == null) {
+        continue;
+      } else {
+        allNull = false;
+      }
+      // Check for invalid input
+      if (num instanceof Byte) {
+        throw new DirectiveExecutionException("Input cannot be of type 'Byte'.");
+      }
+      // Compute average (no BigDecimal case)
+      if (noBigDecimal && num instanceof BigDecimal) {
+        if (t > 1) {
+          avg = BigDecimal.valueOf((double) avg);
+        } else {
+          avg = BigDecimal.ZERO;
+        }
+        noBigDecimal = false;
+      } else if (noBigDecimal) {
```

Comment:
I think this statement makes more sense if it was the `else` of the next `if` block. This makes it obvious that these are the 2 paths the code takes (BigDecimal mode vs Double mode)

Reply:
Makes sense. Let me change that.

Reply:
Changed. Thanks!